pkg/operator/webhooksupportabilitycontroller/webhook_supportability_controller.go

```diff
@@ -59,6 +66,18 @@ func (c *webhookSupportabilityController) sync(ctx context.Context, controllerCo
 		return nil
 	}
 
+	// do nothing while an upgrade is in progress
+	clusterVersion, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		return err
+	}
+	desired := clusterVersion.Status.Desired.Version
+	history := clusterVersion.Status.History
+	// upgrade is in progress if there is no history, or the latest history entry matches the desired version and is not completed
+	if len(history) == 0 || history[0].Version != desired || history[0].State != configv1.CompletedUpdate {
```

Comment:
The `desired` condition is probably overkill.  It's fine if you stay `Degraded` in any window where the CVO has accepted the retarget but not yet written a new `history` entry.  But it doesn't hurt to have the `desired` check either, if you want to keep it.